radsat-sk/operation/subsystems/RPdb.c

```diff
@@ -0,0 +1,272 @@
+/**
+ * @file RPdb.c
+ * @date February 21, 2022
+ * @author Isaac Poirier (iap992)
+ */
+
+#include <RPdb.h>
+#include <RI2c.h>
+#include <string.h>
+
+
+
+/***************************************************************************************************
+                                            DEFINITIONS
+***************************************************************************************************/
+/**
+ * Conversion Constants for between ADC Values and Real Values
+ */
+
+/** Used in: (Volts = Constant * ADC Count) */
+#define ADC_COUNT_TO_BCR_OUTPUT_VOLTAGE			((float) 0.008993157)
+#define ADC_COUNT_TO_BCR_VOLTAGE				((float) 0.0322581)
+#define ADC_COUNT_TO_BATTERY_BUS_OUTPUT_VOLTAGE	((float) 0.008978)
+#define ADC_COUNT_TO_5V_BUS_OUTPUT_VOLTAGE		((float) 0.005865)
+#define ADC_COUNT_TO_3V3_BUS_OUTPUT_VOLTAGE		((float) 0.004311)
+
+
+/** Used in: (mAmps = Constant * ADC Count) */
+#define ADC_COUNT_TO_BCR_OUTPUT_CURRENT			((float) 14.662757)
+
+/** Used in: (Amps = Constant * ADC Count) */
+#define ADC_COUNT_TO_BCR_AMPERAGE				((float) 0.0009775)
+#define ADC_COUNT_TO_3V3_PDB_CURRENT_DRAW		((float) 0.001327547)
+#define ADC_COUNT_TO_5V_PDB_CURRENT_DRAW		((float) 0.001327547)
+#define ADC_COUNT_TO_BATTERY_BUS_OUTPUT_CURRENT	((float) 0.005237)
+#define ADC_COUNT_TO_5V_BUS_OUTPUT_CURRENT		((float) 0.005237)
+#define ADC_COUNT_TO_3V3_BUS_OUTPUT_CURRENT		((float) 0.005237)
+
+
+/** Used in: (Temp in Deg C = (Scale * ADC Count) - Shift) */
+#define ADC_COUNT_TO_SOLAR_ARRAY_TEMP_SCALE	((float) 0.4963)
+// Can use this \/ interchangeably between solar array and motherboard temp shift
+#define ADC_COUNT_TO_MOTHERBOARD_TEMP_SHIFT ((float) 273.15)
+#define ADC_COUNT_TO_MOTHERBOARD_TEMP_SCALE ((float) 0.372434)
+
+
+/** Used in: (Irradiance = Constant * ADC Count) */
+#define ADC_COUNT_TO_IRRADIANCE				((float) 1.59725)
+
+
+/**
+ * Constant Values for I2C Communication
+ */
+/** I2C Slave Address for PDB */
+#define PDB_I2C_SLAVE_ADDR 			(0x2B)
+/** Data Sent to the PDB via I2C for telemetry commands is 6 Bytes, 2 for the command, 4 for data*/
+#define PDB_TELEM_COMMAND_LENGTH	(6)
+/** Data sent to the PDB via I2C for all other commands is 4 Bytes, 2 for the command, 2 for data*/
+#define PDB_COMMAND_LENGTH			(4)
+/** Data Returned from the PDB via I2C is 4 Bytes */
+#define PDB_RESPONSE_LENGTH			(4)
+/** Delay between telemetry read/write operations with the PDB on I2C is 5ms */
+#define PDB_I2C_TELEM_DELAY			(5)
+/** Delay between other read/write operations with the PDB on I2C is 1ms */
```

Comment:
Can you share wherein the datasheets you found this time constraint for the delays

Reply:
Page 43 of the EPS Datasheet. All telemetry calls are given a delay in ms in the document

![Telemetry Table with Delay times](https://user-images.githubusercontent.com/34425321/158466925-5608ed43-1913-4c66-9476-be334311d0ef.png)